rectools/models/random.py

```diff
@@ -71,18 +83,11 @@ def _recommend_u2i(
         if filter_viewed:
             user_items = dataset.get_user_item_matrix(include_weights=False)
 
-        if sorted_item_ids_to_recommend is not None:
-            item_ids = np.unique(sorted_item_ids_to_recommend)
-        else:
-            item_ids = self.all_item_ids
-
-        item_indices = list(range(item_ids.size))  # for random.sample
-
-        np.random.seed(self.random_state)
-        random.seed(self.random_state, version=2)
+        item_ids = self._get_filtered_item_ids(sorted_item_ids_to_recommend)
+        sampler = _RandomSampler(item_ids, self.random_state)
```

Comment:
We will have the same recommendations for multiple `recommend` calls. Is this our desired behaviour? I'm not sure.

Reply:
It's a good point to think.

Actually we had the same behaviour before, so I haven't changed anything. 
But probably it'd be really better to change this.

Reply:
fixed

Reply:
Probably should add this to Changelog